### PR TITLE
Expand array in json root

### LINKED ISSUE #4307: Support expanding array with json root
## Enhancement
Now, the SR >= version 2.1 cannot support expanding json sub-array with json root.
```
{
	"data": [{
		"key": 1
	}, {
		"key": 2
	}]
}
```
We cannot get `key: 1`, `key: 2` by setting json root `$.data` and `strip_outer_array`.
The reason is that the order of the json parsing process is:
1. expanding the array
1. extract json with json root
1. scan json

To parse the json above with json root, we could change the order of process as:
1. extract json with json root
1. expanding the array
1. scan json

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4307 

## Problem Summary(Required) ：

```
{
	"data": [{
		"key": 1
	}, {
		"key": 2
	}]
}
```
This PR adds support for the parse of json showed above, with json root `$.data` and `strip_outer_array=true`.

### CODE REVIEW

**Review by murphyatwork — DISMISSED** [review]
LGTM

File: be/src/exec/vectorized/json_scanner.cpp [hunk 1]
```diff
@@ -781,19 +733,19 @@ Status JsonArrayParser::get_current(simdjson::ondemand::object* row) {
         simdjson::ondemand::value val = *_array_itr;
 
         if (val.type() != simdjson::ondemand::json_type::object) {
-            return Status::DataQualityError("nested array is not supported");
+            return Status::DataQualityError("json object is expected");
```
Comment (meegoo) [inline be/src/exec/vectorized/json_scanner.cpp:736]: This error message is less accurate than the previous one

Comment (rickif) [reply be/src/exec/vectorized/json_scanner.cpp]: Updated. The content of the value is added.

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by rickif** [comment]
@Mergifyio backport branch-2.2

**Comment by rickif** [comment]
@mergifyio backport branch-2.2

**Comment by rickif** [comment]
@Mergifyio backport branch-2.1

**Pull request merged**